### PR TITLE
dev: add remote container debug configs

### PR DESCRIPTION
**Pull request by desimone**
## Summary

Adds remote container development configurations for VS code. These are most useful for step-debugging / developing against other fronting proxies / forward-auth configurations. 

## Related issues

It's related to all my issues. All of them. 

- https://github.com/pomerium/pomerium/pull/1447
- https://github.com/pomerium/pomerium/pull/1415
- #1313 

**Checklist**:

- [x] ready for review

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// For format details, see https://aka.ms/vscode-remote/devcontainer.json or this file's README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.137.0/containers/go
+{
+  "name": "pomerium-remote-containers",
+  "runArgs": ["--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined"],
+  // Set *default* container specific settings.json values on container create.
```
Comment (calebdoxsey) [inline .devcontainer/devcontainer.json:6]: does this work? JSON doesn't allow comments

**Review by calebdoxsey — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (travisgroth) [reply .devcontainer/devcontainer.json:6]: I think VScode config is technically HOCON or something along those lines.  It supports trailing commas, which is great.

File: .gitignore [hunk 2]
```diff
@@ -43,7 +43,6 @@ _testmain.go
 
 .DS_Store
 .idea
-.vscode
```
Comment (travisgroth) [inline .gitignore:46]: Should we leave this in here and just force the commit on this PR?  Otherwise, any local customization will be a commit candidate.

Comment (desimone) [reply .gitignore]: yes

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (desimone) [reply .devcontainer/devcontainer.json:6]: https://code.visualstudio.com/docs/languages/json#_json-with-comments

**Pull request merged**